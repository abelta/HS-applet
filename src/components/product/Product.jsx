import React from 'react';
import PropTypes from 'prop-types';

const Product = ({ order, name, onClick, value }) => (
  <article>
    <button onClick={onClick}>
      <span>{order}</span>
      <span>{name}</span>
      <span>{value}</span>
    </button>
  </article>
);

Product.defaultProps = {
  onClick: null,
};

Product.propTypes = {
  order: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  value: PropTypes.string.isRequired,
};

export default Product;
