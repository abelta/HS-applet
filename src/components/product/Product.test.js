import React from 'react';
import renderer from 'react-test-renderer';
import Product from './Product';

describe('Product', () => {
  const tree = renderer.create(<Product name="name" order={1} value="value" />).toJSON();

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
