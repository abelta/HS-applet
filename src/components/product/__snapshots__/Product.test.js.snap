// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<button
  className="col s10 row btn-flat waves-effect waves-dark white"
  onClick={null}
  style={
    Object {
      "border": "none",
      "margin": 0,
    }
  }
>
  <span
    className="col s2"
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    1
  </span>
  <span
    className="col s5"
  >
    name
  </span>
  <span
    className="col s5"
  >
    value
  </span>
</button>
`;
